state/application_test.go

```diff
@@ -3706,6 +3706,70 @@ func (s *CAASApplicationSuite) TestWatchScale(c *gc.C) {
 	wc.AssertNoChange()
 }
 
+func (s *CAASApplicationSuite) TestRewriteStatusHistory(c *gc.C) {
+	st := s.Factory.MakeModel(c, &factory.ModelParams{
+		Name: "caas-model",
+		Type: state.ModelTypeCAAS, CloudRegion: "<none>",
+	})
+	defer st.Close()
+	f := factory.NewFactory(st, s.StatePool)
+	ch := f.MakeCharm(c, &factory.CharmParams{Name: "gitlab", Series: "kubernetes"})
+	app := f.MakeApplication(c, &factory.ApplicationParams{Name: "gitlab", Charm: ch})
+
+	history, err := app.StatusHistory(status.StatusHistoryFilter{Size: 10})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(history, gc.HasLen, 1)
+	c.Assert(history[0].Status, gc.Equals, status.Waiting)
+	c.Assert(history[0].Message, gc.Equals, "waiting for container")
+
+	// History will use operator status (and not written again.)
+	err = app.SetStatus(status.StatusInfo{
+		Status:  status.Maintenance,
+		Message: "app message",
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	history, err = app.StatusHistory(status.StatusHistoryFilter{Size: 10})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(history, gc.HasLen, 1)
+	c.Assert(history[0].Status, gc.Equals, status.Waiting)
+	c.Assert(history[0].Message, gc.Equals, "waiting for container")
+
+	// Must overwrite the history
+	err = app.SetOperatorStatus(status.StatusInfo{
+		Status:  status.Allocating,
```

Comment:
this won't typically be allocating but more an error condition due to say a bad image

Reply:
If it has issues pulling down the image (i.e. in my testing poisoning the OPERATOR_IMAGE value) the pod status is 'Pending' (which we translate to Allocating) with the message something like "Error: ....".

Reply:
ok